src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java

```diff
@@ -408,14 +445,26 @@ public UserDTO getTrialAccountActivationInfo(@RequestParam String key) {
         throw new CustomMessageRuntimeException("No key found");
     }
 
-
     @PostMapping(path = "/account/resend-verification")
-    public void resendVerification(@RequestBody LoginVM loginVM) {
-        Optional<User> userOptional = userService.getUserWithAuthoritiesByLogin(loginVM.getUsername());
-
-        if (userOptional.isPresent() && passwordEncoder.matches(loginVM.getPassword(), userOptional.get().getPassword())) {
-            mailService.sendActivationEmail(userMapper.userToUserDTO(userOptional.get()));
+    public void resendVerification(@RequestBody LoginVM loginVM, HttpServletRequest request) throws Exception {
+        try {
+            RecaptchaEnterpriseServiceClient client = createAssess.createClient();
+            String recaptchaToken = createAssess.getRecaptchaToken(request);
+            ResponseEntity<String> rs = createAssess.createAssessment(client,recaptchaToken);
+        } catch (ValidationException e) {
+            e.printStackTrace();
+            String errorMessage = e.getMessage();
+            if (errorMessage.contains("Unable to retrieve recaptcha token.")) {
+                throw new Exception(errorMessage);
+            } 
```

Comment:
Is there a way for us to not throw exception until we gather enough info whether token is attached on every request?

Reply:
I believe the recaptcha enterprise console would only provide information about the scores from each request and not the specific error for why the request would fail but I can double check